columnflow/tasks/production.py

```diff
@@ -84,8 +91,13 @@ def run(self):
         output_chunks = {}
 
         # run the producer setup
-        producer_reqs = self.producer_inst.run_requires()
-        reader_targets = self.producer_inst.run_setup(producer_reqs, luigi.task.getpaths(producer_reqs))
+        self._array_function_post_init()
+        producer_reqs = self.producer_inst.run_requires(task=self)
+        reader_targets = self.producer_inst.run_setup(
+            task=self,
+            reqs=producer_reqs,
+            inputs=luigi.task.getpaths(producer_reqs),
+        )
```

Comment:
are the `post_init` only being run directly before the requires func or are they also invoked outside of a run method sometimes?
I am thinking a bit on the use cases like using the `btag_weights.PRODUCES` in the `uses` of another CSP, does that still work?
Example: https://github.com/uhh-cms/hh2bbww/blob/c88c9bca0e72d7e6fd2b9b4da155624ff168bfd0/hbw/production/normalized_btag.py#L21

Reply:
Yes, that still works.